instrumentation/grpc-1.5/src/main/java/io/opentelemetry/instrumentation/hypertrace/grpc/v1_5/Http2HeadersInstrumentationModule.java

```diff
@@ -0,0 +1,133 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.hypertrace.grpc.v1_5;
+
+import static net.bytebuddy.matcher.ElementMatchers.failSafe;
+import static net.bytebuddy.matcher.ElementMatchers.isMethod;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArguments;
+
+import com.google.auto.service.AutoService;
+import io.grpc.Metadata;
+import io.netty.handler.codec.http2.Http2Headers;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge;
+import io.opentelemetry.javaagent.tooling.InstrumentationModule;
+import io.opentelemetry.javaagent.tooling.TypeInstrumentation;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.hypertrace.agent.core.HypertraceSemanticAttributes;
+
+@AutoService(InstrumentationModule.class)
+public class Http2HeadersInstrumentationModule extends InstrumentationModule {
+
+  public Http2HeadersInstrumentationModule() {
+    super(GrpcInstrumentationName.PRIMARY, GrpcInstrumentationName.OTHER);
+  }
+
+  @Override
+  public List<TypeInstrumentation> typeInstrumentations() {
+    return Arrays.asList(new NettyUtilsInstrumentation());
+  }
+
+  /**
+   * The server side HTTP2 headers are added in tracing gRPC interceptor. The headers are added to
+   * metadata in {@link GrpcUtils_convertHeaders_Advice}.
+   *
+   * <p>The client side HTTP2 headers are added directly to span in {@link
+   * Utils_convertClientHeaders_Advice}. TODO However it does not work for the first request
+   * https://github.com/hypertrace/javaagent/issues/109#issuecomment-740918018.
+   */
+  class NettyUtilsInstrumentation implements TypeInstrumentation {
+    @Override
+    public ElementMatcher<? super TypeDescription> typeMatcher() {
+      return failSafe(named("io.grpc.netty.Utils"));
+    }
+
+    @Override
+    public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+      Map<ElementMatcher<? super MethodDescription>, String> transformers = new HashMap<>();
+      transformers.put(
+          isMethod().and(named("convertClientHeaders")).and(takesArguments(6)),
+          Utils_convertClientHeaders_Advice.class.getName());
+      transformers.put(
+          isMethod().and(named("convertHeaders")).and(takesArguments(1)),
+          GrpcUtils_convertHeaders_Advice.class.getName());
+      return transformers;
+    }
+  }
+
+  static class Utils_convertClientHeaders_Advice {
```

Comment:
nit: GrpcUtils maybe?

Reply:
The target class is called `Utils` therefore the `Utils` name.